src/GRA.Controllers/MissionControl/EmailManagementController.cs

```diff
@@ -207,5 +227,97 @@ public async Task<IActionResult> SendEmail(EmailIndexViewModel viewModel)
                 });
             }
         }
+
+        public async Task<IActionResult> Addresses()
+        {
+            var user = await _userService.GetDetails(GetActiveUserId());
+            if (user.IsAdmin)
+            {
+                var signupData = _emailManagementService.GetSignUpSourcesWithCount();
+                var viewModel = new EmailAddressesViewModel
+                {
+                    IsAdmin = user.IsAdmin,
+                    SignUpSources = new SelectList(signupData, "Value", "DisplayText")
+                };
+                return View(viewModel);
+            }
+            ShowAlertDanger("Page not found.");
+            return RedirectToAction(nameof(EmailManagementController.Index));
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> ExportAddresses(EmailAddressesViewModel viewModel)
+        {
+            var user = await _userService.GetDetails(GetActiveUserId());
+            if (user.IsAdmin)
+            {
+                if (string.IsNullOrEmpty(viewModel.SignUpSource))
+                {
+                    ShowAlertDanger("Could not export Email Reminders");
+                    ModelState.AddModelError(nameof(viewModel.SignUpSource), "The signup source is required.");
+                    var signupData = _emailManagementService.GetSignUpSourcesWithCount();
+                    viewModel.IsAdmin = user.IsAdmin;
+                    viewModel.SignUpSources = new SelectList(signupData, "Value", "DisplayText");
+                    return View("Addresses",viewModel);
+                }
+                var ms = new MemoryStream();
+                var writer = new StreamWriter(ms);
+                var data = _emailManagementService.GetEmailRemindersBySignUpSource(viewModel.SignUpSource);
+                var json = JsonConvert.SerializeObject(data.Select(_ => new { _.CreatedAt, _.Email, _.SignUpSource }));
+                writer.WriteLine(json);
+                writer.Flush();
+                ms.Seek(0, SeekOrigin.Begin);
+                var fileOutput = new FileStreamResult(ms, "text/plain")
+                {
+                    FileDownloadName = $"Email_Reminders_{DateTime.Now.ToString()}.txt"
+                };
+                return fileOutput;
+            }
+            ShowAlertDanger("Page not found.");
+            return RedirectToAction(nameof(EmailManagementController.Index));
+        }
+        
+        [HttpPost]
+        public async Task<IActionResult> ImportAddresses(EmailAddressesViewModel viewModel)
+        {
+            var user = await _userService.GetDetails(GetActiveUserId());
+            if (user.IsAdmin)
+            {
+                if (viewModel.UploadedFile == null)
+                {
+                    ShowAlertDanger("Could not import Email Reminders");
```

Comment:
On error this should set environment stuff like the alert danger and then do a redirect to view rather than returning the view from here - the Post/Redirect/Get pattern